03 Writing Algorithms/03 Securities/05 Filtering Data/99 Examples.html

```diff
@@ -1,103 +1,122 @@
-<h4>Example 1: Only Within 1 Hour Of Market Open And Close</h4>
-<p>Some traders prefer to use only data of 1 hour after market open and before market close, since they have the most volume.</p>
+<h4>Example 1: Filter Price Outliners Using SMA</h4>
+<p>When analyzing high-frequency price data, it's important to filter out potential outliers and anomalies that may skew the analysis. One effective method is to use a 1-month SMA to identify price ticks that deviate significantly from the short-term trend. By comparing each price tick to the 1-month SMA, you can flag any data points that fall outside a specified threshold, which we used 3 standard deviations in this example. This helps remove suspicious or erroneous price information, ensuring a cleaner dataset for further technical analysis and trading strategies.</p>
 <div class="section-example-container">
-	<pre class="csharp">public override void Initialize()
+	<pre class="csharp">public class SampleAlgorithm : QCAlgorithm
 {
-    var security = AddEquity("SPY");
-    security.SetDataFilter(new CustomDataFilter());
-}
+    public override void Initialize()
+    {
+        dynamic equity = AddEquity("AAPL", Resolution.Daily);
+        // Set a 22-step for 1 month daily price SMA
+        equity.Sma = SMA(equity.Symbol, 22, Resolution.Daily);
+        // Set 22-step rolling window of the SMA indicator to save last month data to calculate standard deviation
+        equity.Sma.Window.Size = 22;
+        // Warm up the indicator
+        WarmUpIndicator((Symbol)equity.Symbol, (SimpleMovingAverage)equity.Sma, Resolution.Daily);
+        // Set the data filter
+        equity.SetDataFilter(new CustomDataFilter());
+    }
```

Comment:
Redo for tick data. Filtering daily bars isn't something people do. Always create realistic examples.